crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -1086,11 +1088,41 @@ def evacuate_network_node(controller, network_node, delete_namespaces = false)
         )
         Rails.logger.info("Migrating loadbalancers away from #{hostname} was successful.")
 
+        args = [hostname, "disable"]
+        save_node_action("disabling network agents")
+        controller.wait_for_script_to_finish(
+          "/usr/sbin/crowbar-set-network-agents-state.sh",
+          timeouts[:set_network_agents_state],
+          args
+        )
+        Rails.logger.info("Disabling network agents on #{hostname} was successful.")
+
         network_node.run_ssh_cmd("mkdir -p /var/lib/crowbar/upgrade; touch #{migrated_file}")
         # Cleanup up the ok/failed state files, as we likely need to
         # run the script again on this node (to evacuate other nodes)
         controller.delete_script_exit_files("/usr/sbin/crowbar-router-migration.sh")
         controller.delete_script_exit_files("/usr/sbin/crowbar-lbaas-evacuation.sh")
+        controller.delete_script_exit_files("/usr/sbin/crowbar-set-network-agents-state.sh")
+      rescue StandardError => e
+        raise_node_upgrade_error(
+          e.message +
+          " Check /var/log/crowbar/node-upgrade.log at #{controller.name} for details."
+        )
+      end
+
+      # Re-enable neutron network agents on a just upgraded node
+      def re_enable_network_agents(controller, network_node)
```

Comment:
Wouldn't it make sense to put this method into `api/node.rb` ?

Reply:
Hm, this method is somewhat similar to `evacuate_network_node` in that is calls something on a Node that is different from the node that is currently being upgrade. And currently not all callers of this method have a node_api object from the target node available. If we want to move this to api/node.rb we should move `evacuate_network_node` as well.